loqusdb/plugins/mongo/structural_variant.py

```diff
@@ -134,12 +134,92 @@ def add_structural_variant(self, variant, max_window = 3000):
                 }
             }
         )
-        
+
         # Insert an identity object to link cases to variants and clusters
-        identity_obj = Identity(cluster_id=cluster['_id'], variant_id=variant['id_column'], 
+        identity_obj = Identity(cluster_id=cluster['_id'], variant_id=variant['id_column'],
                                 case_id=case_id)
         self.db.identity.insert_one(identity_obj)
-        
+
+        return
+
+    def delete_structural_variant(self, variant, max_window = 3000):
+
+        # This will return the cluster most similar to variant or None
+        cluster = self.get_structural_variant(variant)
+
+        if cluster is None:
+            return
+
+        case_id = variant.get('case_id')
+
+        if case_id in cluster['families']:
+            # Insert the new case in the beginning of array
+            cluster['families'].remove(case_id)
+
+        nr_events = cluster['observations'] - 1
+
+        if nr_events == 0:
+            # Remove SV if no observations remain
+            self.db.structural_variant.delete_one({'_id': cluster['_id']})
+
+            # remove identiry
+            identity_obj = Identity(cluster_id=cluster['_id'], variant_id=variant['id_column'],
+                                    case_id=case_id)
+            self.db.identity.delete_one(identity_obj)
+
+            return
+
+        pos_mean = int((cluster['pos_sum'] - variant['pos']) // (nr_events))
```

Comment:
Could you put this in helper functions since the same code is used in at least two places? It would also be nice to test this on its own